modules/queue/src/main/scala/ch/epfl/bluebrain/nexus/service/queue/KafkaPublisher.scala

```diff
@@ -0,0 +1,83 @@
+package ch.epfl.bluebrain.nexus.service.queue
+
+import java.util.concurrent.Future
+
+import akka.NotUsed
+import akka.actor.{ActorRef, ActorSystem}
+import akka.kafka.ProducerMessage._
+import akka.kafka.ProducerSettings
+import akka.kafka.scaladsl.Producer
+import akka.persistence.query.Offset
+import akka.stream.scaladsl.Flow
+import cats.Show
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.service.indexer.persistence.SequentialTagIndexer
+import io.circe.Encoder
+import io.circe.syntax._
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import shapeless.Typeable
+
+/**
+  * Helper class to send messages to different Kafka topics sharing only one instance
+  * of a [[org.apache.kafka.clients.producer.KafkaProducer]]
+  *
+  * @param producer the shared [[KafkaProducer]] instance
+  * @param topic    the kafka topic
+  * @tparam Event   the generic event type
+  */
+class KafkaPublisher[Event: Encoder: Show](producer: KafkaProducer[String, String], topic: String) {
```

Comment:
I think `Show` may be too generic for an `Event` type, could we consider defining a separate type class like `Key[A]` that extracts the key off an `Event`?